src/main/java/org/folio/utils/CalculationUtils.java

```diff
@@ -107,10 +109,10 @@ private static Map<String, BigDecimal> calculate(Double dInitialAllocation,
     BigDecimal awaitingPayment = BigDecimal.valueOf(dAwaitingPayment);
 
     BigDecimal allocated = initialAllocation.add(allocationTo).subtract(allocationFrom);
-    BigDecimal unavailable = encumbered.add(awaitingPayment).add(expended);
+    BigDecimal unavailable = ensureNonNegative(encumbered.add(awaitingPayment).add(expended).subtract(credited));
     BigDecimal totalFunding = allocated.add(netTransfers);
-    BigDecimal cashBalance = totalFunding.subtract(expended);
-    BigDecimal available = totalFunding.subtract(unavailable);
+    BigDecimal cashBalance = totalFunding.subtract(expended).add(credited);
+    BigDecimal available = totalFunding.subtract(encumbered.add(awaitingPayment).add(expended).subtract(credited));
```

Comment:
probably better to add comment here that we do not use unavailable because we dont need logic "positive or zero" here or even better to extract encumbered.add(awaitingPayment).add(expended).subtract(credited) into new variable

Reply:
thanks extracted to variable